src/lib/reasoners/satml_frontend.ml

```diff
@@ -67,6 +67,8 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
     mutable unknown_reason : Sat_solver_sig.unknown_reason option;
     (** The reason why satml raised [I_dont_know] if it does; [None] by
         default. *)
+    declare_ids : Id.typed Vec.t;
+    declare_lim : int Vec.t;
```

Comment:
Uhm, `Id.typed list Vec.t`is a stack, not a queue in your suggestion, right?

Reply:
Yes it's a stack (for some reasons they are called "queues" elsewhere in satML but we can use the proper teminology ;) ).

Reply:
So why do we need an extra field `declare_ids` here? The top element of the stack can be accessed in `O(1)`. I believe 
```ocaml 
declare_stack: Id.typed list Vec.t 
```
is sufficient.

Reply:
Even if the algorithmic complexity is identical, the constant term is not the same; having a separate field here avoids having to check that the vector is not empty on all accesses. It probably does not matter here because we never look at the declarations except during model generation but for the other vectors in SatML which are accessed much more frequently it might matter.

It's fine to use a non-empty `Vec.t` here (or even a `Stack.t` from the stdlib actually).

Reply:
I will use the current design with an extra field.